test/e2e/metrics.go

```diff
@@ -0,0 +1,59 @@
+package e2e
+
+import (
+	"context"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/kubernetes/test/e2e/framework"
+
+	topologyclientset "github.com/k8stopologyawareschedwg/noderesourcetopology-api/pkg/generated/clientset/versioned"
+	"github.com/k8stopologyawareschedwg/resource-topology-exporter/test/e2e/utils"
+	"github.com/onsi/ginkgo"
+	"github.com/onsi/gomega"
+)
+
+const (
+	rteContainerName = "resource-topology-exporter-container"
+)
+var _ = ginkgo.Describe("[RTE] metrics", func() {
+	var (
+		initialized         bool
+	)
+
+	f := framework.NewDefaultFramework("metrics")
+
+	ginkgo.BeforeEach(func() {
+		var err error
+
+		if !initialized {
+			_, err = topologyclientset.NewForConfig(f.ClientConfig())
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+			_, err = f.ClientSet.CoreV1().Nodes().Get(context.TODO(), getNodeName(), metav1.GetOptions{})
```

Comment:
what is this for, if we throw away the result and keep only the error?

Reply:
Yes, the first one is redundant but the second one will hint if the system is ready for test

Reply:
ok, if you need to wait for the nodes to be ready (do we actually need? can't we just run the metrics scraping in a Eventually block and just wait and bail out on error?), better to do that explicitely.

Reply:
On second thought yes, adding a sync point to make sure the nodes are ready gives us a clearer error message. But let's do explicitely.

Reply:
What do you mean by doing that explicitly? something like ```Eventually()```?

Reply:
If we need to check the nodes are ready, for example, let's actually make logic on the node objects instead of only checking the error message of the `Get`, which tells us nothing about the state of the nodes.
The code as-is will only tell us that a given node _exists_.